cluster-autoscaler/processors/provreq/processor_test.go

```diff
@@ -150,12 +151,55 @@ func TestRefresh(t *testing.T) {
 		pr.Status.Conditions = test.conditions
 		pr.CreationTimestamp = metav1.NewTime(test.creationTime)
 		pr.Spec.ProvisioningClassName = v1beta1.ProvisioningClassCheckCapacity
+
 		additionalPr := provreqclient.ProvisioningRequestWrapperForTesting("namespace", "additional")
 		additionalPr.CreationTimestamp = metav1.NewTime(weekAgo)
 		additionalPr.Spec.ProvisioningClassName = v1beta1.ProvisioningClassCheckCapacity
-		processor := provReqProcessor{func() time.Time { return now }, 1, provreqclient.NewFakeProvisioningRequestClient(nil, t, pr, additionalPr), nil}
-		processor.refresh([]*provreqwrapper.ProvisioningRequest{pr, additionalPr})
+
+		oldFailedPr := provreqclient.ProvisioningRequestWrapperForTesting("namespace", "failed")
```

Comment:
IMO this should be a separate test case (which would require reworking this test a bit), or a separate test altogether. As it's right now it adds a lot of complexity to the test code for no apparent reason.